extern/ttconv/pprdrv_tt.cpp

```diff
@@ -800,16 +800,23 @@ void ttfont_sfnts(TTStreamWriter& stream, struct TTFONT *font)
 
     /* Now, generate those silly numTables numbers. */
     sfnts_pputUSHORT(stream, count);            /* number of tables */
-    if ( count == 9 )
-    {
-        sfnts_pputUSHORT(stream, 7);          /* searchRange */
-        sfnts_pputUSHORT(stream, 3);          /* entrySelector */
-        sfnts_pputUSHORT(stream, 81);         /* rangeShift */
-    }
-    else
-    {
-        debug("only %d tables selected",count);
+
+    int search_range = 1;
+    int entry_sel = 0;
+
+    while (search_range <= count) {
+        search_range <<= 1;
+        entry_sel++;
     }
+    entry_sel = entry_sel > 0 ? entry_sel - 1 : 0;
+    search_range = (search_range >> 1) * 16;
+    int range_shift = count * 16 - search_range;
+
+    sfnts_pputUSHORT(stream, search_range);      /* searchRange */
+    sfnts_pputUSHORT(stream, entry_sel);         /* entrySelector */
+    sfnts_pputUSHORT(stream, range_shift);       /* rangeShift */
+
+    debug("only %d tables selected",count);
```

Comment:
Not sure that this debug message is needed. I'm guessing it was in the code to warn in the case that there were fewer than 9 tables, in which case the header was the wrong length.